simpleclient/src/main/java/io/prometheus/client/Gauge.java

```diff
@@ -181,6 +181,21 @@ public void setToCurrentTime() {
     public Timer startTimer() {
       return new Timer(this);
     }
+
+    /**
+     * Executes runnable code (ie Java 8 Lambda) and observes a duration of how long it took to run.
+     *
+     * @param timeable Code that is being timed
+     */
+    public void time(Runnable timeable){
```

Comment:
I'm not sure on the naming here, as we use setDuration as against observeDuration for the existing timers. This is to reduce the chances that someone will incorrectly use this where they should use a Summary/Historgram.

`setToTime` maybe?